packages/playwright/src/runner/taskRunner.ts

```diff
@@ -105,9 +105,14 @@ export class TaskRunner<Context> {
       status = 'failed';
     }
     cancelPromise?.resolve();
-    // Note that upon hitting deadline, we "run cleanup", but it exits immediately
-    // because of the same deadline. Essentially, we're not performing any cleanup.
-    const cleanup = () => teardownRunner.runDeferCleanup(context, deadline).then(r => r.status);
+    const cleanup = async () => {
+      // Upon hitting deadline we add extra 30s to actually perform cleanup, otherwise
```

Comment:
Let's update the documentation to state this explicitly.

Reply:
Updated.